### PR TITLE
Update docstrings of `qml.equal` and `qml.assert_equal`

### PR DESCRIPTION
**Pull request by astralcai**
**Context:**
Some contents of the `qml.equal` docstring is outdated.

**Description of the Change:**
Updates the `qml.equal` docstrings

### CODE REVIEW

File: pennylane/ops/functions/equal.py [hunk 1]
```diff
@@ -56,38 +56,33 @@ def equal(
...
 
-        The ``qml.equal`` function is based on a comparison of the type and attributes
-        of the measurement or operator, not a mathematical representation. While
-        comparisons between some classes, such as ``Tensor`` and ``Hamiltonian``, are
-        supported, mathematically equivalent operators defined via different classes
-        may return False when compared via ``qml.equal``.
-
-        To be more thorough would require the matrix forms to be calculated, which may
-        drastically increase runtime.
+        The ``qml.equal`` function is based on a comparison of the types and attributes of the
```
Comment (albi3ro) [inline pennylane/ops/functions/equal.py:63]: How is this block changed? Hard to notice the difference.

File: pennylane/ops/functions/equal.py [hunk 2]
```diff
@@ -56,38 +56,33 @@ def equal(
...
 
-        The kwargs ``check_interface`` and ``check_trainability`` can only be set when
-        comparing ``Operation`` objects. Comparisons of ``MeasurementProcess``
-        or ``Observable`` objects will use the default value of ``True`` for both, regardless
-        of what the user specifies when calling the function. For subclasses of ``SymbolicOp``
-        or ``CompositeOp`` with an ``Operation`` as a base, the kwargs will be applied to the base
-        comparison.
+        The interfaces and trainability of data within some observables including ``Tensor``,
+        ``Hamiltonian``, ``Prod``, ``Sum`` are ignored, regardless of what the user specifies
+        for ``check_interface`` and ``check_trainability``.
```
Comment (albi3ro) [inline pennylane/ops/functions/equal.py:74]: Are we really ignoring this for `Prod` and `Sum`? 

Seems like it matters here:
```
>>> op1 = qml.RX(qml.numpy.array(0.1), 0) + qml.X(0)
>>> op2 = qml.RX(0.1, 0) + qml.X(0)
>>> qml.equal(op1, op2)
False
>>> qml.equal(op1, op2, check_interface=False, check_trainability=False)
False
```

Comment (astralcai) [reply pennylane/ops/functions/equal.py:74]: I wasn't testing with `RX`, see:
```pycon
>>> qml.equal(
    qml.X(0) + qml.s_prod(0.1, qml.Y(1)),
    qml.X(0) + qml.s_prod(qml.numpy.array(0.1), qml.Y(1))
)
True
```

File: pennylane/ops/functions/equal.py (hunk 1 above)
Comment (astralcai) [reply pennylane/ops/functions/equal.py:63]: Mostly some minor changes in wording:

type and attributes -> types and attributes
measurement or operator -> measurements or operators
not a mathematical representation -> not their mathematical representations
while comparisons between some classes, ..., are supported -> while mathematical comparisons ...

File: pennylane/ops/functions/equal.py (hunk 2 above)
Comment (astralcai) [reply pennylane/ops/functions/equal.py:74]: I did some investigation, and I find that the interfaces and trainability are ignored only when the two operators both have a `pauli_rep` and their `pauli_rep`s equal.
```pycon
>>> qml.equal(
...    qml.Hamiltonian([0.1], [qml.X(0)]),
...    qml.Hamiltonian([qml.numpy.array(0.1)], [qml.X(0)]),
... )
True

>>> qml.equal(
...    qml.Hamiltonian([0.1], [qml.RX(0.1, wires=0)]),
...    qml.Hamiltonian([qml.numpy.array(0.1)], [qml.RX(0.1, wires=0)]),
... )
False

>>> qml.equal(
...    qml.Hamiltonian([0.1], [qml.RX(0.1, wires=0)]),
...    qml.Hamiltonian([qml.numpy.array(0.1)], [qml.RX(0.1, wires=0)]),
...    check_interface=False,
...    check_trainability=False,
... )
True
```

**Review by albi3ro — APPROVED** [review]

**Review by AmintorDusko — APPROVED** [review]
Nice!

**Pull request merged**